test/e2e/app-dir/segment-cache/incremental-opt-in/segment-cache-incremental-opt-in.test.ts

```diff
@@ -45,6 +46,19 @@ describe('segment cache (incremental opt in)', () => {
             } else {
               prefetches.set(key, prefetchInfo)
             }
+            const response = await page.request.fetch(request, {
+              maxRedirects: 0,
+            })
+            const status = response.status()
+            if (status !== 200) {
+              unexpectedResponse.push({
+                status,
+                url,
+                headers: response.headers(),
+                response: await response.text(),
+              })
+            }
```

Comment:
Collecting the unexpected responses allows the test to fail early instead of running into a timeout.